### PR TITLE
[release-v0.18] Fix VirtualMachineCRCErrors not stop firing

### PR DESCRIPTION
**Pull request by machadovilaca**
This is a manual cherry-pick of https://github.com/kubevirt/ssp-operator/pull/742

https://issues.redhat.com/browse/CNV-35848

**Release note**:

```release-note
Fix VirtualMachineCRCErrors not stop firing
```

### CODE REVIEW

**Comment by 0xFelix** [comment]
@machadovilaca Looks like the e2e test you introduced with this PR is failing?

File: tests/monitoring_test.go [hunk 1]
```diff
@@ -287,7 +286,7 @@ var _ = Describe("Prometheus Alerts", func() {
 
 			Eventually(func(g Gomega) error {
 				return apiClient.Get(ctx, types.NamespacedName{Name: vmName, Namespace: strategy.GetNamespace()}, vm)
-			}).Should(MatchError(k8serrors.IsNotFound))
```
Comment (0xFelix) [inline tests/monitoring_test.go]: IIRC this syntax is just missing a string as second argument

e.g.

`Should(MatchError(k8serrors.IsNotFound, "IsNotFound"))`

Comment (machadovilaca) [reply tests/monitoring_test.go]: that is what we have in main branch, but that caused the automatic cherry-pick to fail. Probably here we have an older version of gomega

Comment (0xFelix) [reply tests/monitoring_test.go]: ACK

**Review by 0xFelix — APPROVED** [review]
/approve
/lgtm

File: tests/monitoring_test.go [hunk 2]
```diff
@@ -287,7 +286,7 @@ var _ = Describe("Prometheus Alerts", func() {
...
+			}).Should(MatchError(ContainSubstring(fmt.Sprintf("virtualmachines.kubevirt.io \"%s\" not found", vmName))))
```
Comment (akrejcir) [inline tests/monitoring_test.go:289]: The default timeout for `Eventually()` is 1 second, which is probably not enough in this case.
Can you increase it? And the default polling interval is 10ms, which is too low.
https://pkg.go.dev/github.com/onsi/gomega#Eventually 

I should have commented on: https://github.com/kubevirt/ssp-operator/pull/742 , but I didn't notice it.

Comment (akrejcir) [reply tests/monitoring_test.go:289]: @machadovilaca , please open a new PR with this change.

Comment (machadovilaca) [reply tests/monitoring_test.go:289]: https://github.com/kubevirt/ssp-operator/pull/913

**Pull request merged**